### PR TITLE
Pull in the new template utils with copy template functions

### PR DESCRIPTION
**Pull request by gparvin**
Adding support for the new go template functions:
- copyConfigMapData
- copySecretData These copy the data section of ConfigMap or Secret resources so individual keys don't have to be specified

Refs:
 - https://issues.redhat.com/browse/ACM-2611

### CODE REVIEW

File: test/resources/case13_templatization/case13_copy_referenced_configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    include: ["default"]
+  object-templates:
+    - complianceType: musthave
+      objectDefinition:
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: configmap-copy-configmap-object-repl
+          namespace: default
+        data: '{{ copyConfigMapData "default" "configmap-copy-configmap-object" | autoindent }}'
```
Comment (mprahl) [inline test/resources/case13_templatization/case13_copy_referenced_configmap.yaml:18]: Is `autoindent` necessary?

File: test/e2e/case13_templatization_test.go [hunk 2]
```diff
@@ -324,4 +367,53 @@ var _ = Describe("Test templatization", func() {
...
+
+			By("By verifying that the replicated ConfigMap has the expected data")
+			replConfigMap, err := clientManaged.CoreV1().ConfigMaps("default").Get(
+				context.TODO(), configMapReplName, v1.GetOptions{},
+			)
+			Expect(err).To(BeNil())
+			Expect(replConfigMap.Data["message"]).To(Equal("Hello Raleigh!"))
+		})
+		It("Cleans up", func() {
+			deleteConfigPolicies([]string{case13CopyRefObject})
```
Comment (yiraeChristineKim) [inline test/e2e/case13_templatization_test.go:414]: Is this clean up working well without "ordered", ginkgo runs parallel sometimes it disturbs other testings

File: test/e2e/case13_templatization_test.go [hunk 3]
```diff
@@ -324,4 +367,53 @@ var _ = Describe("Test templatization", func() {
...
+			utils.Kubectl("delete", "configmap", configMapName, "-n", "default")
+			utils.Kubectl("delete", "configmap", configMapReplName, "-n", "default")
```
Comment (yiraeChristineKim) [inline test/e2e/case13_templatization_test.go:416]: recommend add "--ignore-not-found"

Comment (mprahl) [reply test/e2e/case13_templatization_test.go:416]: Yes, good idea!

File: test/resources/case13_templatization/case13_copysecret.yaml [hunk 4]
```diff
@@ -15,6 +15,4 @@ spec:
         metadata:
           name: e2esecret2
         type: Opaque
-        data:
-          USER_NAME: YWRtaW4=
-          PASSWORD: '{{ fromSecret "default" "e2esecret" "PASSWORD" }}'
+        data: '{{ copySecretData "default" "e2esecret" | autoindent }}'
```
[the remark this answers is not in the document]
Comment (gparvin) [reply test/resources/case13_templatization/case13_copysecret.yaml]: no, I removed it.  Thanks

File: test/e2e/case13_templatization_test.go (hunk 3 above)
Comment (gparvin) [reply test/e2e/case13_templatization_test.go]: I added it.  Thanks

File: test/e2e/case13_templatization_test.go [hunk 5]
```diff
@@ -324,4 +367,53 @@ var _ = Describe("Test templatization", func() {
...
+			).Should(Equal("Compliant"))
+
+			By("By verifying that the replicated ConfigMap has the expected data")
+			replConfigMap, err := clientManaged.CoreV1().ConfigMaps("default").Get(
+				context.TODO(), configMapReplName, v1.GetOptions{},
+			)
+			Expect(err).To(BeNil())
+			Expect(replConfigMap.Data["message"]).To(Equal("Hello Raleigh!"))
+		})
+		It("Cleans up", func() {
```
Comment (mprahl) [inline test/e2e/case13_templatization_test.go:413]: This should ideally be an `AfterAll`, but it looks like the rest of the test code in this file does it this way already. For the sake of time, I'll let this slide, but in the future, please use `AfterAll`.

**Review by mprahl — APPROVED** [review]

**Pull request merged**